pkg/common/vclib/datacenter.go

```diff
@@ -40,12 +40,27 @@ type Datacenter struct {
 // GetDatacenter returns the DataCenter Object for the given datacenterPath
 // If datacenter is located in a folder, include full path to datacenter else just provide the datacenter name
 func GetDatacenter(ctx context.Context, connection *VSphereConnection, datacenterPath string) (*Datacenter, error) {
-	finder := find.NewFinder(connection.Client, false)
-	datacenter, err := finder.Datacenter(ctx, datacenterPath)
-	if err != nil {
-		klog.Errorf("Failed to find the datacenter: %s. err: %+v", datacenterPath, err)
-		return nil, err
+	var datacenter *object.Datacenter
+	var err error
+
+	dcRef := object.ReferenceFromString(datacenterPath)
```

Comment:
The name of the function is not that explicit. Could you add a comment to say check if the datacenter path is moid format or not?

Reply:
done